crates/sui-core/src/gateway_types.rs

```diff
@@ -694,14 +714,24 @@ impl From<MoveStruct> for SuiMoveStruct {
 #[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, Eq, PartialEq)]
 #[serde(rename = "MovePackage")]
 pub struct SuiMovePackage {
+    modules: BTreeMap<String, Vec<u8>>,
     disassembled: BTreeMap<String, Value>,
 }
 
+impl SuiMovePackage {
+    pub fn modules(&self) -> &BTreeMap<String, Vec<u8>> {
+        &self.modules
+    }
+}
+
 impl TryFrom<MoveModulePublish> for SuiMovePackage {
     type Error = anyhow::Error;
 
+    // TODO: this is a bug--the disassembled code is going to have the wrong package ID (0x0 instead of the ID generated by the tx)
```

Comment:
@patrickkuo @tnowacki seeking your advice on how to handle this. I think we probably need to eliminate this API and create the `SuiMovePackage` from the result of a publish tx (which will have the correct object ID's) instead of from a `MoveModulePublish` (which unfortunately just doesn't have enough info).